core/backend/src/CheckpointManager.ts

```diff
@@ -215,7 +215,7 @@ export class V2CheckpointManager {
         const maxRequests = getPrefetchConfig("maxRequests", 6);
         const timeout = getPrefetchConfig("timeout", 100);
         const maxBlocks = getPrefetchConfig("maxBlocks", 500); // default size of 2GB. Assumes a checkpoint block size of 4MB.
-        if (dbStats?.totalBlocks !== undefined && dbStats.totalBlocks <= maxBlocks) {
+        if (dbStats?.totalBlocks !== undefined && dbStats.totalBlocks <= maxBlocks && dbStats.nPrefetch === 0) {
```

Comment:
I'm confused. When does nPrefetch get incremented/decremented?

Reply:
nPrefetch is incremented in the C file blockcachevfsd.c. https://github.com/iTwin/imodel-native/commit/5b690f078020cfef3d0c577f4af4974a718d900c#diff-e028b1b8c73d077614fd355e57805a59af03233b6af64eed725aa21d09dd632aR1792